src/service/dashboards.rs

```diff
@@ -12,78 +12,107 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use actix_web::{
-    http::{self, StatusCode},
-    HttpResponse,
-};
+use actix_web::{http::StatusCode, HttpResponse};
 use std::io;
-use tracing::info_span;
+use tracing::instrument;
 
 use crate::meta::{self, dashboards::Dashboard, http::HttpResponse as MetaHttpResponse};
 use crate::service::db::dashboard;
 
-pub async fn get_dashboard(org_id: &str, name: &str) -> Result<HttpResponse, io::Error> {
-    let loc_span = info_span!("service:dashboards:get");
-    let _guard = loc_span.enter();
+#[derive(Debug)]
+enum Response {
+    Created { dashboard_id: String },
+    NotFound { error: Option<String> },
+    InternalServerError(anyhow::Error),
+}
 
-    match dashboard::get(org_id, name).await {
-        Ok(Some(dashboard)) => Ok(HttpResponse::Ok().json(dashboard)),
-        Ok(None) => Ok(HttpResponse::NotFound().json(MetaHttpResponse::error(
-            StatusCode::NOT_FOUND.into(),
-            "Dashboard not found".to_string(),
-        ))),
-        Err(_) => Ok(HttpResponse::NotFound().json(MetaHttpResponse::error(
-            StatusCode::NOT_FOUND.into(),
-            "Dashboard not found".to_string(),
-        ))),
+impl From<Response> for HttpResponse {
+    fn from(resp: Response) -> Self {
+        match resp {
+            Response::Created { dashboard_id } => Self::Created().json(MetaHttpResponse::message(
+                StatusCode::CREATED.into(),
+                format!("Dashboard {dashboard_id} created"),
+            )),
+            Response::NotFound { error } => Self::NotFound().json(MetaHttpResponse::error(
+                StatusCode::NOT_FOUND.into(),
+                error.unwrap_or("Dashboard not found".to_owned()),
+            )),
+            Response::InternalServerError(err) => Self::InternalServerError().json(
+                MetaHttpResponse::error(StatusCode::INTERNAL_SERVER_ERROR.into(), err.to_string()),
+            ),
+        }
     }
 }
 
-pub async fn save_dashboard(
+#[instrument(skip(dashboard))]
+pub async fn create_dashboard(
     org_id: &str,
-    name: &str,
-    dashboard: &Dashboard,
+    dashboard: Dashboard,
 ) -> Result<HttpResponse, io::Error> {
-    let loc_span = info_span!("service:dashboards:save");
-    let _guard = loc_span.enter();
+    // XXX-TODO(vvv): Generate[*] `dashboard_id` and overwrite
+    // `dashboard.dashboard_id` with this value.
+    //
+    // [*]: Use global counter, stored in the database. Increment the counter
+    // on each `create_dashboard` call.
```

Comment:
@oasisk Does this approach look reasonable?

Reply:
FWIIW, here's a list of possible approaches (courtesy of ChatGPT-3):
> 1. UUID - A Universally Unique Identifier (UUID) is an identifier standard used in software construction. Pros: The data is unique across all systems and is difficult to guess. Cons: It is a lengthy identifier and may not be suitable for some applications.
> 2. Auto-Increment ID - Auto-incrementing IDs are generated by incrementing a number for each new record. Pros: Easy to generate, short, and simple. Cons: Not secure since the IDs can be guessed and it is difficult to enforce uniqueness across multiple systems.
> 3. Hash-based ID - A hash-based ID is generated by applying a cryptographic hash function to the data. Pros: Secure, and the IDs cannot be guessed. Cons: Longer identifiers, and can be difficult to generate.

🤖